### DISCUSSION TITLE
AM2302 Temperature and Humidity module

### DISCUSSION THREAD

**Discussion by Baggypants** [bkbilly/lnxlink#81] [Show and tell]
I wrote a pretty basic module to get data out of a temperature and humidity module and put it in to Homeassistant.

Pics are here
https://mstdn.social/@Baggypants/111426451981872218

A terrible repo hosting the module is here:

https://gitlab.com/Baggypants/temp-and-humidity-a2308-module-for-lnxlink

**Comment by bkbilly** [comment]
This looks very useful!
LNXlink has a function that automates the library installation, which can be used on your module.
```python
from .scripts.helpers import import_install_package
adafruit_lib = import_install_package("adafruit-circuitpython-dht", ">=4.0.0", "adafruit_dht")
dhtDevice = adafruit_lib.DHT22(board.D4)
```

Take a look at the [GPU](https://github.com/bkbilly/lnxlink/blob/master/lnxlink/modules/gpu.py) module or any other that uses the `_requirements` method.
